imsim/diffraction.py

```diff
@@ -42,7 +42,61 @@ class Geometry:
 )
 
 
-def diffraction_kick(
+def apply_diffraction_delta(
+    pos: np.ndarray,
+    v: np.ndarray,
+    t: np.ndarray,
+    wavelength: np.ndarray,
+    lat: float,
+    telescope_orientation: np.ndarray,
+    geometry: Geometry,
+    distribution: Callable[[np.ndarray], np.ndarray],
+) -> np.ndarray:
+    """Statistically diffract photons.
+
+    For a ray of photons with positions pos, wavelengths wavelength and pupil plane
+    coordinates pos,
+    randomly generate diffraction angles and change the directions of the photons
+    entering the pupil plane.
+
+    Parameters
+    ----------
+    pos : 2d positions of the intersections of the rays with the pupil plane (shape (n, 2)).
+    v : Direction of the photons (shape (n, 3)).
```

Comment:
This actually comes in with shape (3,n).  Personally, I think the code would be clearer if this *was* (n,3) as the doc string says, since right now I find the transposes required below and the way that v is indexed to be rather confusing.  But right now the doc string and the code don't match.

Reply:
Sorry, actually we still need to feed `v` directly to `batoid.RayVector`. Therefore, for performance reasons, I would recommend to go with shape (3,n)

Reply:
You can make the array column-major.  Use `order='F'` when making the array.  Then it can be (n,3) but still have contiguous columns.

Reply:
I used `np.array(...).T` this also leads to a column major array.